amethyst_network/src/server/host.rs

```diff
@@ -0,0 +1,100 @@
+/// This module is about the communication whit this host to some other endpoints.
+///
+/// 1. Sending Data
+/// 2. Receiving Data
+/// 3. Broadcasting
+use crate::error::Result;
+use crate::server::{
+    ReceiveHandler, SendHandler, ServerConfig, ServerSocketEvent, UdpReceiver, UdpSender,
+};
+use laminar::{DeliveryMethod, Packet};
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+/// 'Host' abstracts TCP and UDP sockets away.
+pub struct Host {
+    config: ServerConfig,
+    // Handler to access the internals of the UDP receiving thread
+    udp_receiver: ReceiveHandler,
+    // Handler to access the internals of the UDP sender thread
+    udp_sender: SendHandler,
+    // Handler to access the internals of the TCP receiving thread
+    tcp_receiver: Option<ReceiveHandler>,
+    // Handler to access the internals of the TCP sender thread
+    tcp_sender: Option<Arc<SendHandler>>,
+}
+
+impl Host {
+    /// This will start and return an instance of the host.
+    /// 1: Fire up a TCP-sender and TCP-receiver if enabled in config.
+    /// 2: Fire up a UDP-sender and UDP-receiver.
+    /// 3: Set up some `channels` to communicate with underlying threads.
+    ///
+    /// The method uses the config provided when creating a `host` instance.
+    pub fn run(config: ServerConfig) -> Result<Host> {
+        if config.enable_tcp {
+            // todo: create TCP support.
```

Comment:
Can we print out a message here so the user doesn't think TCP is working when it really isn't?